src/hawkbit-client.c

```diff
@@ -806,7 +812,8 @@ static gpointer download_thread(gpointer data)
         process_deployment_cleanup();
         g_mutex_unlock(&active_action->mutex);
 
-        return NULL;
+        *thread_ret = FALSE;
+        return g_steal_pointer(&thread_ret);
```

Comment:
This also means we always have to free the return of `g_thread_join()` right? Currently in the unref before (re)creating this thread we don't do that.

Reply:
Please take a look at the new changes. I guess that's not needed anymore, right?